### ISSUE TITLE
Negative Jacobian potentially due to a problem with velocity solver

### ISSUE THREAD

**Issue by echoi** [sungho91/Laghost#2]
LAGHOST built with the latest MFEM (commit 9343ff) is failing to run with `./defaults.cfg`.
The working commit ID of MFEM is `c0867ca009e9b5fafb5585318648f430b1d6eef0`.
The nominal reason is `dt=0`, which in turn is directly caused by a negative Jacobian:
In `void LagrangianGeoOperator::UpdateQuadratureData(const Vector &S) const`,
 
```C++
if (min_detJ < 0.0)
    {
       // This will force repetition of the step with smaller dt.
       qdata.dt_est = 0.0;
    }
    else
    {
       if (inv_dt>0.0)
       {
          qdata.dt_est = fmin(qdata.dt_est, cfl*(1.0/inv_dt));
          qdata.h_est  = fmin(qdata.h_est, h_min);
       }
    }
```
At this point, it has been found that the nodal coordinates, the first block of the solution block vector is corrupted when `void LagrangianGeoOperator::SolveVelocity(const Vector &S, Vector &dS_dt) const` is called for the first time. A more fine-grained analysis on this function is still needed. 

For simplicity, the analysis was conducted with a much simpler mesh and input parameters than the default ones. The used files are attached.
[defaults_EC.txt](https://github.com/user-attachments/files/16488540/defaults_EC.txt)
[2d_mesh_mfem_example.mesh.txt](https://github.com/user-attachments/files/16488541/2d_mesh_mfem_example.mesh.txt)

**Comment by echoi** [comment]
This issue has been fixed by the commit e5863ff5.

**Comment by echoi** [comment]
This issue has been fixed by the commit e5863ff5.
